### DISCUSSION TITLE
Roujin's

### DISCUSSION THREAD

**Discussion by Tigertye** [CorsixTH/CorsixTH#2235] [Research Room]
I saw that Roujin's code is listed as working now and I wondered what exactly is it supposed to do? 

There's almost no info out there on it but I gather it's a spawn rate thing. I assumed this meant I would eventually get a horde at my hospital. How many game years does that take?

I activated Roujin's when I got the first fax at my hospital so it's been running for almost 5 game years now. I really don't see any difference at this point. The first 6 months or so yes, there was a difference and it was awesome but the spawn rate never got better <or worse depending on your perspective>. 5 years on and the spawn rate seems the same as it did at 6 months.

Is there any way to turn it up?

I'm playing with the Biggen map. There are no population controls in the level file and the only thing I have added to it is #emergency_control[0].Mean.Variance 365 90

**Comment by SilPho** [comment]
As you've guessed, the Roujin cheat does increase the spawn rate, but it does this by setting it at 40 patients per month (give or take a few), every month.

By comparison, the default spawn behaviour for custom levels, which I'm guessing is what Biggen uses, starts off at 4 patients per month and slowly ramps up to a maximum of 30. For reference, the campaign levels from the original game have spawn rate limits that seem to range from about 16 to 40, depending on the level.

If you want a bigger challenge you could adjust the population controls in an existing map file, or create your own custom level.

**Reply by Tigertye** [reply]
It's limited to 40ish? Oh that's not near enough for the Biggen map. Well, adjusting the population controls isn't an issue at all but it does mean I need to start over. I wish it were possible to tweak the controls in an existing save. I have been trying to conquer this map for several years now. Every time I adjust the level file I have to start a new build to get the changes to take affect. Oh well, if I haven't given up by now I doubt I ever will :)

Thanks for responding.
TT

**Comment by lewri** [comment]
Roujin is a static spawn rate. It's still tied to population share and reputation I think. So 1000 rep can make the monthly visitors go even higher (200%?).

The reason we don't document too much about the cheats is because they're meant to be somewhat hidden. The wiki itself has basic details though.

**Reply by lewri** [reply]
https://github.com/CorsixTH/CorsixTH/wiki/Cheats

**Reply by Tigertye** [reply]
i found this when searching which is why i wondered if it got better eventually: 
https://github.com/CorsixTH/CorsixTH/blob/master/CorsixTH/Lua/world.lua

1260    local no_of_spawns = math.n_random(self.spawn_rate, 2)
1261    -- If Roujin's Challenge is on, override spawn rate

So would Roujin's also override the spawn rate in a custom level file or add to the spawn in that instance?

**Reply by SilPho** [reply]
@lewri Good point about reputation/population (I'm still learning how everything works).

Given that hospital population starts at 0.25 (25%), it means that this line...
`no_of_spawns = math.ceil(no_of_spawns*self:getLocalPlayerHospital().population)`
...suggests that 10 patients per month is what you actually expect to see to start with.

Hospital population, as far as I can tell, maxes out at 75% of the spawn rate, so even in a perfect hospital the highest you'd see is 40 * 0.75 = 30 patients per month.

I did a quick test with a brand new hospital and activating the cheat does seem to attract exactly 10 per month (starting at the end of the current month).

@Tigertye You're almost in the right spot. That's how the spawning normally works, but just below that is where the Roujin cheat overrides the result of that calculation. Turns out what I said before about "40 (give or take a few)" is wrong, it's just 40 if the cheat is enabled. However, as I just said that's actually likely to be cut down to between 10 and 30 per month instead.

**Reply by Tigertye** [reply]
If I am understanding this correctly then I would do better to adjust the population controls in my level file rather than using Roujin's which is severely curtailing the spawn rate compared to what I want to happen. So maybe something like this?

#popn[0].Month.Change 0 4
#popn[1].Month.Change 1 1
#popn[1].Month.Change 24 15
#popn[1].Month.Change 25 0
#popn[1].Month.Change 36 15
#popn[1].Month.Change 37 0
#popn[1].Month.Change 48 15
#popn[1].Month.Change 49 0
#popn[1].Month.Change 60 20
#popn[1].Month.Change 61 0
#popn[1].Month.Change 72 20
#popn[1].Month.Change 73 0
#popn[1].Month.Change 84 20
#popn[1].Month.Change 85 0
#popn[1].Month.Change 96 20
#popn[1].Month.Change 97 0
#popn[1].Month.Change 108 20
#popn[1].Month.Change 109 0
#popn[1].Month.Change 120 1

**Reply by SilPho** [reply]
If you want more than the Roujin cheat then yes, I believe that's the route you need to go down.

I will add that most levels seem to leave the increase value at 0 so that there is an upper limit but feel free to leave it uncapped for the brave souls that go beyond 10 years 😁

**Reply by Tigertye** [reply]
LOL
I'll let you know if I blow up the hospital even though that is kind of my goal ;)

**Reply by lewri** [reply]
190 patients at month 120.... Gosh. Increasing by one each month thereafter.
It's likely the experience would become unenjoyable at such high numbers.

**Reply by Tigertye** [reply]
I've always been curious as to how much a hospital can truly handle. I'm kinda weird that way. 
I think I finally have a good layout and build plan. My biggest frustration is having to start over every time I tweak the level file.

**Comment by Alberth289346** [comment]
Make the cheat relative to the current population (eg `1.25 * current_popn` or so), perhaps with a minimum increase?
